discord/application_command.go

```diff
@@ -435,3 +444,131 @@ func (c MessageCommand) CreatedAt() time.Time {
+	applicationID            snowflake.ID
+	guildID                  *snowflake.ID
+	name                     string
+	nameLocalizations        map[Locale]string
+	nameLocalized            string
+	defaultMemberPermissions Permissions
+	dmPermission             bool
+	nsfw                     bool
+	integrationTypes         []ApplicationIntegrationType
+	contexts                 []InteractionContextType
+	version                  snowflake.ID
+	handler                  *EntryPointCommandHandlerType
+}
+
+func (c *EntryPointCommand) UnmarshalJSON(data []byte) error {
+	var v rawEntryPointCommand
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
+	}
+
+	c.id = v.ID
+	c.applicationID = v.ApplicationID
+	c.guildID = v.GuildID
+	c.name = v.Name
+	c.nameLocalizations = v.NameLocalizations
+	c.nameLocalized = v.NameLocalized
+	c.defaultMemberPermissions = v.DefaultMemberPermissions
+	c.dmPermission = v.DMPermission
+	c.nsfw = v.NSFW
+	c.integrationTypes = v.IntegrationTypes
+	c.contexts = v.Contexts
+	c.version = v.Version
+	c.handler = v.Handler
+	return nil
+}
+
+func (c EntryPointCommand) MarshalJSON() ([]byte, error) {
+	return json.Marshal(rawEntryPointCommand{
+		ID:                       c.id,
+		Type:                     c.Type(),
+		ApplicationID:            c.applicationID,
+		GuildID:                  c.guildID,
+		Name:                     c.name,
+		NameLocalizations:        c.nameLocalizations,
+		NameLocalized:            c.nameLocalized,
+		DefaultMemberPermissions: c.defaultMemberPermissions,
+		DMPermission:             c.dmPermission,
+		NSFW:                     c.nsfw,
+		IntegrationTypes:         c.integrationTypes,
+		Contexts:                 c.contexts,
+		Version:                  c.version,
+		Handler:                  c.handler,
+	})
+}
+
+func (c EntryPointCommand) ID() snowflake.ID {
+	return c.id
+}
+
+func (EntryPointCommand) Type() ApplicationCommandType {
+	return ApplicationCommandTypePrimaryEntryPoint
+}
+
+func (c EntryPointCommand) ApplicationID() snowflake.ID {
+	return c.applicationID
+}
+
+func (c EntryPointCommand) GuildID() *snowflake.ID {
+	return c.guildID
+}
+
+func (c EntryPointCommand) Name() string {
+	return c.name
+}
+
+func (c EntryPointCommand) NameLocalizations() map[Locale]string {
+	return c.nameLocalizations
+}
+
+func (c EntryPointCommand) NameLocalized() string {
+	return c.nameLocalized
+}
+
+func (c EntryPointCommand) DefaultMemberPermissions() Permissions {
+	return c.defaultMemberPermissions
+}
+
+func (c EntryPointCommand) DMPermission() bool {
+	return c.dmPermission
+}
+
+func (c EntryPointCommand) NSFW() bool {
+	return c.nsfw
+}
+
+func (c EntryPointCommand) IntegrationTypes() []ApplicationIntegrationType {
+	return c.integrationTypes
+}
+
+func (c EntryPointCommand) Contexts() []InteractionContextType {
+	return c.contexts
+}
+
+func (c EntryPointCommand) Version() snowflake.ID {
+	return c.version
+}
+
+func (c EntryPointCommand) Handler() *EntryPointCommandHandlerType {
```

Comment:
does this need to be a method? we usually have fields which aren't part of the interface as public fields